### PR TITLE
Set up testing environment for API

### PR DESCRIPTION
**Pull request by mauroavellaneda**
[PR](https://www.pivotaltracker.com/story/show/175016869)
## This PR contains:
- Adds gems capybara, factory_bot_rails, rspec-rails and shoulda-matchers.
- Creates support folder. 
- Creates file 'shoulda_matchers.rb'. 
- Configure FactoryBot and shoulda matchers in rails helper.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -2,19 +2,21 @@ source 'https://rubygems.org'
...
-gem 'rails', '~> 6.0.3', '>= 6.0.3.3'
+gem 'bootsnap', '>= 1.4.2', require: false
 gem 'pg', '>= 0.18', '< 2.0'
 gem 'puma', '~> 4.1'
-gem 'bootsnap', '>= 1.4.2', require: false
+gem 'rails', '~> 6.0.3', '>= 6.0.3.3'
 
 group :development, :test do
-  gem 'byebug', platforms: %i[mri mingw x64_mingw]
+  gem 'capybara'
```
Comment (emtalen) [inline Gemfile]: For what do we use capybara? Do we need it for testing an API?

**Review by emtalen — CHANGES_REQUESTED** [review]
Please set up the reponse JSON module as well 👍🏻

File: Gemfile (hunk 1 above)
Comment (mauroavellaneda) [reply Gemfile]: Removed.

Comment (emtalen) [reply Gemfile]: Yes I can see that, but that was not my question.

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -20,4 +20,5 @@
   config.filter_rails_from_backtrace!
   config.include FactoryBot::Syntax::Methods
   config.include(Shoulda::Matchers::ActiveRecord, type: :model)
+  config.include ResponseJSON
```
Comment (emtalen) [inline spec/rails_helper.rb:23]: You need to add the actual module to the support folder

**Review by emtalen — CHANGES_REQUESTED** [review]

**Review by emtalen — APPROVED** [review]

**Pull request merged**